chia/rpc/wallet_rpc_api.py

```diff
@@ -937,21 +937,23 @@ async def get_transactions(self, request: Dict[str, Any]) -> EndpointResult:
         tx_list = []
         # Format for clawback transactions
         for tr in transactions:
-            try:
-                tx = (await self._convert_tx_puzzle_hash(tr)).to_json_dict_convenience(self.service.config)
-                tx_list.append(tx)
-                if tx["type"] not in CLAWBACK_INCOMING_TRANSACTION_TYPES:
-                    continue
-                coin: Coin = tr.additions[0]
-                record: Optional[WalletCoinRecord] = await self.service.wallet_state_manager.coin_store.get_coin_record(
-                    coin.name()
-                )
-                assert record is not None, f"Cannot find coin record for type {tx['type']} transaction {tx['name']}"
-                tx["metadata"] = record.parsed_metadata().to_json_dict()
-                tx["metadata"]["coin_id"] = coin.name().hex()
-                tx["metadata"]["spent"] = record.spent
-            except Exception:
-                log.exception(f"Failed to get transaction {tr.name}.")
+            tx = (await self._convert_tx_puzzle_hash(tr)).to_json_dict_convenience(self.service.config)
+            tx_list.append(tx)
+            if tx["type"] not in CLAWBACK_INCOMING_TRANSACTION_TYPES:
+                continue
+            coin: Coin = tr.additions[0]
+            record: Optional[WalletCoinRecord] = await self.service.wallet_state_manager.coin_store.get_coin_record(
+                coin.name()
+            )
+            if record is None:
+                log.error(f"Cannot find coin record for type {tx['type']} transaction {tx['name']}")
+                continue
+            if record.metadata is None:
```

Comment:
I hesitate a bit with this precheck.  Can we not leave this detail to `.parsed_metadata()` and handle when it fails?

Reply:
We could use, it raises so I suppose it would be pythonic to let it do so inside a try/except

I think letting the standard RPC handling for exceptions is good for anything outside the typical flow - so those would be all reported in the log and back to the caller

Reply:
The main point of not pre-checking is to avoid writing the checks twice.  The second place will almost certainly not get maintained and then still in some case result in an exception anyways.

Also, note my general comment.  Previous code continued the loop on 'unexpected' exceptions, but was silent.  This code is noisy (in a good way), but doesn't continue.  Maybe we want noisy and also continuing?  Don't let one failure cost you from finding all other valid records.

Reply:
No, I want to short-circuit on things very strange and only continue for specific cases around clawback handling. See if  the latest commit works for you